osgp/platform/osgp-adapter-kafka-distributionautomation/src/main/java/org/opensmartgridplatform/adapter/kafka/da/infra/kafka/out/LowVoltageMessageProducer.java

```diff
@@ -96,16 +100,20 @@ public void send(final String measurement) {
     private ScadaMeasurementPayload addLocationData(final ScadaMeasurementPayload[] payloads) {
         final ScadaMeasurementPayload payload = payloads[0];
         final String substationIdentification = payload.getSubstationIdentification();
-        payload.setSubstationName(this.locationConfig.getSubstationLocation(substationIdentification));
-        final String feeder = payload.getFeeder();
+
+        final Optional<Location> locationOptional = this.locationService.getLocation(substationIdentification);
+        payload.setSubstationName(locationOptional.map(Location::getName).orElse(""));
```

Comment:
Usage of `Optional` is best left as return type for methods that might not have a return value.
I don't see a real need for an `Optional` variable here as the following alternative looks fine (I think) without going against the way optionals were intended:
`final String substationName = this.locationService.getLocation(substationIdentification).map(Location::getName).orElse("");`
`payload.setSubstationName(substationName);`

Given that the optional is used again later, an alternative could be to refactor to a method taking a Location and call that from ifPresent attached on the getLocation return value.

As we are lacking proper conventions for using Optionals in a certain way, and not in others, you can let me know whether you agree (and update the code) or not.